core/src/main/scala/org/apache/spark/shuffle/FetchFailedException.scala

```diff
@@ -68,5 +68,6 @@ private[spark] class FetchFailedException(
 private[spark] class MetadataFetchFailedException(
     shuffleId: Int,
     reduceId: Int,
-    message: String)
-  extends FetchFailedException(null, shuffleId, -1L, -1, reduceId, message)
+    message: String,
+    cause: Throwable = null)
```

Comment:
I don't see `cause` is used anywhere. Shall we covert the `cause` to stack string and append to the `message`?

Reply:
Sure, I just want to keep original stack trace.